### PR TITLE
feat(theme): display default gateway forms

### PR DESCRIPTION
**Pull request by ashraf706**
Shows default gateway forms if custom forms are not present.
This pull request also includes main.css and roboto font files.
The location of these files may be changed as required.

#### Issue: https://github.com/gravitee-io/issues/issues/8315

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/forgot_password.html [hunk 1]
```diff
@@ -46,7 +46,7 @@
                    th:name="${bot_detection_configuration.get('tokenParameterName')}" />
 
             <div>
-                <button type="submit" id="submitBtn" class="button primary">Sent</button>
+                <button type="submit" id="submitBtn" class="button submit-button">Send</button>
```
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/forgot_password.html:49]:
```suggestion
                <button type="submit" id="submitBtn" class="button primary">Send</button>
```

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/test/java/io/gravitee/am/management/handlers/management/api/resources/FormsResourceTest.java [hunk 2]
```diff
@@ -59,6 +58,7 @@ public void shouldGetForm() throws IOException {
         mockForm.setReferenceId(domainId);
 
         doReturn(Maybe.just(mockForm)).when(formService).findByDomainAndTemplate(domainId, Template.LOGIN.template());
+        doReturn(Single.just(mockForm)).when(formService).getDefaultByDomainAndTemplate(domainId, Template.LOGIN.template());
```
Comment (leleueri) [inline gravitee-am-management-api/gravitee-am-management-api-rest/src/test/java/io/gravitee/am/management/handlers/management/api/resources/FormsResourceTest.java:61]: There is a missing TestCase, here `findByDomainAndTemplate` return a value so the result of `getDefaultByDomainAndTemplate` isn't used to generate the response.

We should have the same test but with `findByDomainAndTemplate` that return `Maybe.emtpy`

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/FormServiceImpl.java [hunk 3]
```diff
@@ -297,6 +301,24 @@ public Completable delete(String domain, String formId, User principal) {
...
+        return Single.create(emitter -> {
+            try {
+                final String path = "views/default/" + template + ".html";
+                final InputStream resourceAsStream = this.getClass().getClassLoader().getResourceAsStream(path);
+                final String content = IOUtils.toString(resourceAsStream, defaultCharset());
+                final Form form = new Form(false, template);
+                form.setContent(content);
+                emitter.onSuccess(form);
+            } catch (Exception e) {
+                emitter.onError(new TechnicalManagementException("An error has occurred while trying load default form", e));
```
Comment (leleueri) [inline gravitee-am-service/src/main/java/io/gravitee/am/service/impl/FormServiceImpl.java:317]:
```suggestion
                emitter.onError(new TechnicalManagementException("An error has occurred while trying to load default form", e));
```

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/FormServiceImpl.java [hunk 4]
```diff
@@ -297,6 +301,24 @@ public Completable delete(String domain, String formId, User principal) {
         return delete(ReferenceType.DOMAIN, domain, formId, principal);
     }
 
+    @Override
+    public Single<Form> getDefaultByDomainAndTemplate(String domain, String template) {
+        LOGGER.debug("Default form by domain {} and template {}", domain, template);
+
+        return Single.create(emitter -> {
+            try {
+                final String path = "views/default/" + template + ".html";
```
Comment (leleueri) [inline gravitee-am-service/src/main/java/io/gravitee/am/service/impl/FormServiceImpl.java:310]: Open question:

Should we introduce theme or domain in the directory tree to highlight the fact that these templates are the defaults coming from the gateway ?

**Review by leleueri — CHANGES_REQUESTED** [review]
In addition of the

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/FormServiceImpl.java (hunk 3 above)
Comment (ashraf706) [reply gravitee-am-service/src/main/java/io/gravitee/am/service/impl/FormServiceImpl.java:317]: done

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/FormServiceImpl.java (hunk 4 above)
Comment (ashraf706) [reply gravitee-am-service/src/main/java/io/gravitee/am/service/impl/FormServiceImpl.java:310]: the `default` folder is created as the `view` folder contains a `login.html`. The sub-folder structure could be any appropriate directories. I have no reservation.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/forgot_password.html (hunk 1 above)
Comment (ashraf706) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/forgot_password.html:49]: done

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/test/java/io/gravitee/am/management/handlers/management/api/resources/FormsResourceTest.java (hunk 2 above)
Comment (ashraf706) [reply gravitee-am-management-api/gravitee-am-management-api-rest/src/test/java/io/gravitee/am/management/handlers/management/api/resources/FormsResourceTest.java:61]: new test added as suggested

**Comment by ashraf706** [comment]
> In addition of the comments here after, can you explain to me why you introduced the assets into the UI ?

Possibly we have already discussed it after standup among three of us. Happy to explain it again and then we can decide whether we can get the job done without these assets in the UI.

**Review by leleueri — APPROVED** [review]

**Review by leleueri — CHANGES_REQUESTED** [review]
The new method introduced into the FormServiceImpl has no Unit Test.
Should at least test :
* Nominal case
* Case where the template is unknown

**Pull request merged**